src/tmqh-flow.c

```diff
@@ -272,6 +277,33 @@ void TmqhOutputFlowIPPair(ThreadVars *tv, Packet *p)
     return;
 }
 
+static void TmqhOutputFlowFTPHash(ThreadVars *tv, Packet *p)
+{
+    uint32_t qid;
+    TmqhFlowCtx *ctx = (TmqhFlowCtx *)tv->outctx;
+
+    if (p->flags & PKT_WANTS_FLOW) {
+        uint32_t hash = p->flow_hash;
+        if (p->tcph != NULL && ((p->sp > 1024 && p->dp > 1024) || p->dp == 21 || p->sp == 21)) {
```

Comment:
should this be `>= 1024`?

also, should this include port 20?

Reply:
> should this be `>= 1024`?

Right

> also, should this include port 20?

Not sure : if it is port 20, it is ftp-data without expectation, right ?

Reply:
No, active ftp mode also uses expectations.

Reply:
2 SV tests using port 20: `bug-4877` and `filestore-v2.7-stream-depth`